data/transactions/logic/assembler.go

```diff
@@ -1531,6 +1531,30 @@ func (ops *OpStream) trackStack(args StackTypes, returns StackTypes, instruction
 	}
 }
 
+// processFields walks through the input fields until it gets to a semi-colon
+// at the end of a field at which point it returns everything prior as current
+// and everything following the semicolon as next
+
+// fields should not be used after as processFields mangles it
+// current and next do share the same array if next is not nil
+func processFields(fields []string) (current, next []string) {
+	for i := 0; i < len(fields); i++ {
+		field := fields[i]
+		if string(field[len(field)-1]) == ";" {
+			field = field[0 : len(field)-1]
+			current = fields[:i]
+			if len(field) > 0 {
+				current = append(current, field)
+			}
+			if i+1 == len(fields) {
+				return current, nil
+			}
+			return current, fields[i+1:]
+		}
+	}
```

Comment:
I think you can use `for i, field := range fields {`.
At first i thought you couldn't, but since once you find something you stop iterating and return, you don't have to worry about moving things around in `fields`

`strings.HasSuffix(field, ";")` will also improve readability